src/plugins/mpi/pmix/mpi_pmix.c

```diff
@@ -89,6 +89,12 @@ const uint32_t plugin_version = SLURM_VERSION_NUMBER;
 
 void *libpmix_plug = NULL;
 
+static pthread_t _abort_tid = 0;
+
+char abort_ip[255] = "";
+int abort_port = -1;
+static pthread_mutex_t abort_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Place SRUN endpoint information (IP/port) & FD in the pmixp_info database instead of global var's (reuse the fields from pmixp info, currently utilized in stepd).